mergify_engine/github_events.py

```diff
@@ -289,7 +298,7 @@ async def _get_github_pulls_from_sha(client, repo, sha, pulls):
     elif pull_number == -1:
```

Comment:
@sileht I think there was a bug here because AFAIU `redis.get` returns a `str` so it's never `== -1`. :(

Reply:
yes, but by chance, since this occurs only when `pull_number is None`, we just don't cache the fact we known that no open pull request have this sha.

Also this code is useless since we don't do GitHub API call in this method anymore. I will remove it.

Reply:
https://github.com/Mergifyio/mergify-engine/pull/1693